osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -190,6 +180,48 @@ private void updateBeatmapDifficulties()
             }
         }
 
+        [Resolved]
+        private BeatmapCarousel.CarouselScrollContainer scrollContainer { get; set; } = null!;
+
+        private void loadContentIfRequired()
+        {
+            Quad containingSsdq = scrollContainer.ScreenSpaceDrawQuad;
+
+            // Using DelayedLoadWrappers would only allow us to load content when on screen, but we want to preload while off-screen
+            // to provide a better user experience.
+
+            // This is tracking time that this drawable is updating since the last pool.
+            // This is intended to provide a debounce so very fast scrolls (from one end to the other of the carousel)
+            // don't cause huge overheads.
+            //
+            // We increase the delay based on distance from centre, so the beatmaps the user is currently looking at load first.
+            float timeUpdatingBeforeLoad = 50 + Math.Abs(containingSsdq.Centre.Y - ScreenSpaceDrawQuad.Centre.Y) / containingSsdq.Height * 100;
+
+            Debug.Assert(Item != null);
+
+            if (loadCancellation == null && (timeSinceUnpool += Time.Elapsed) > timeUpdatingBeforeLoad)
```

Comment:
What on earth is happening here? This is nigh unreadable. What is this doing? What is "unpool"? And is it intentional that `timeSinceUnpool` will only be incremented so long as the `loadCancellation == null` conditional is true?

Reply:
I've restructured this to hopefully make more sense.